core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+        'Title length exceeds the allowed' +
+          ` limit of ${maxLength} characters.`
+      );
+    }
+  }
+
+  /**
+   * This function helps in adding a goal.
+   */
+  async updateGoalTo(goal: string): Promise<void> {
+    await this.clickOn(addGoal);
+    await this.type(addGoal, goal);
+  }
+
+  /**
+   * This function checks if the goal has been set in the exploration.
+   */
+  async expectGoalToEqual(expectedGoal: string): Promise<void> {
+    try {
+      const goalInput = await this.page.$('#explorationObjective');
+      if (!goalInput) {
+        throw new Error('Goal input element not found.');
+      }
+
+      const goal = await this.page.evaluate(input => input.value, goalInput);
+
+      if (goal === expectedGoal) {
+        showMessage('The goal has been set for the exploration.');
+      } else {
+        throw new Error('The goal does not match the expected goal.');
+      }
+    } catch (error) {
+      console.error('Error:', error.message);
+      throw error;
+    }
+  }
+
+  /**
+   * This function helps in selecting a category from dropdown.
+   */
+  async selectACategory(category: string): Promise<void> {
+    await this.clickOn(categoryDropDown);
+    await this.clickOn(category);
+  }
+
+  /**
+   * This function checks if a category has been selected for the exploration.
+   */
+  async expectSelectedCategoryToBe(expectedCategory: string): Promise<void> {
+    await this.page.waitForSelector('.mat-select-value');
+    const selectedCategory = await this.page.evaluate(() => {
+      return (
+        document.querySelector('.mat-select-value') as HTMLElement
+      ).innerText.trim();
+    });
+    if (selectedCategory === expectedCategory) {
+      showMessage(
+        `The category ${selectedCategory}` + ' is same as expectedCategory.'
+      );
+    } else {
+      throw new Error('Category is not correct.');
+    }
+  }
+
+  /**
+   * This function helps in selecting language from dropdown.
+   */
+  async selectALanguage(language: string): Promise<void> {
+    await this.page.evaluate(() => {
+      window.scrollTo(0, 350);
+    });
+
+    await this.clickOn(languageUpdateBar);
+    await this.clickOn(language);
+  }
+
+  /**
+   *  This function verifies that the selected language is displayed correctly.
+   */
+  async expectSelectedLanguageToBe(expectedLanguage: string): Promise<void> {
+    await this.page.waitForSelector(languageUpdateBar);
+    const languageDropdown = await this.page.$(languageUpdateBar);
+    if (!languageDropdown) {
+      throw new Error('Category dropdown not found.');
+    }
+    await languageDropdown.click();
+
+    const selectedLanguage = await this.page.evaluate(() => {
+      const matOption = document.querySelector(
+        '.mat-option.mat-selected'
+      ) as HTMLElement;
+      if (!matOption) {
+        throw new Error('Selected language option not found.');
+      }
+      const selectedLanguageText = matOption.innerText.trim();
+
+      /**
```

Comment:
This is a bit too complex logic for the acceptance tests. Can you just check that the given language is a substring of the one in the dropdown (or maybe check that the one in the dropdown ends with `'(' + expectedLanguage + ')'` or whatever it should be)?

Reply:
@seanlip There can be too much languages and P&C so IMO after testing most implementation this is the most optimised one.
In order to make it simple or understandable I have used doc strings also.

Reply:
I don't understand your rationale for why the suggestion I provided isn't a suitable one. (I don't know what P&C means and I don't know what you are basing the claim of "most optimised" on.) Could you please elaborate?

Reply:
Corrected the logic